docroot/profiles/custom/cgov_site/modules/custom/cgov_core/src/Plugin/Block/SectionNav.php

```diff
@@ -179,14 +179,28 @@ public function renderNavElement(NavItem $navItem, int $renderDepth, int $curren
    * {@inheritdoc}
    */
   public function build() {
+
+    $build = [];
+
     $navTree = $this->getSectionNav();
     if ($navTree) {
       $build = [
         '#type' => 'block',
-        'nav_tree' => $navTree,
+        '#nav_tree' => $navTree,
       ];
-      return $build;
+
+      // Set the cache tag if this if a nav root exists for this section.
+      $navRoot = $this->navMgr->getNavRoot('field_section_nav_root');
+      if ($navRoot) {
```

Comment:
Logic for adding the section root as a cache tag.

Reply:
I think you are also going to have to add a context here, right? To tell it to vary based on ID right now... oh right, the max-age is still 0, so it would not get cached anyway.

Also, I have stared at the sun and adding tags to the build array is not actually the best way in a block's build. The build is actually called as a `#pre_render` function AFTER the thing has been rendered. Technically it should be done in getCacheTags().

Reply:
I held off on adding the context since I reverted back to max-age = 0, but I guess it doesn't hurt to add it now.

The plot thickens...I'll give it a look.